internal/versions/versions.go

```diff
@@ -284,3 +264,44 @@ func (h *handler) AddOpenshiftVersion(ocpReleaseImage, pullSecret string) (*mode
 
 	return openshiftVersion, nil
 }
+
+// Returns a string value by a specified field name
+func (h *handler) getRhcosField(openshiftVersion, fieldName string) (string, error) {
```

Comment:
Is there another way we could parse the data without subverting go's strong typing? And especially without using `unsafe`? What's motivating this approach vs. something like unmarshalling json into a struct?

Reply:
That means that any validation currently done in these getters will be done by the accessing code. So it won't be any cleaner than the current situation.
I've pushed a new version without using 'unsafe'. So maybe we could keep the reflection:/

Reply:
What validation are you referring to? The data doesn't change at runtime, so presumably it can be validated once at startup.

Reply:
There are runtime validations, like ensuring that the specified version exists when creating a cluster.

Reply:
Ok, `GetVersion` can either return a models.OpenshiftVersion or an error if one doesn't exist. We're just checking if something exists in a map, right? Seems like standard stuff. Any other validations?

Reply:
Ok, so added a GetOsImage funk for fetching the entire struct, and the rest of the validation is done when creating a new Versions handler. We could do the same for other funcs in the API later, to reduce duplication.